### ISSUE TITLE
Maintainer tools freezes on Privacy table

### ISSUE THREAD

**Issue by IanMayo** [debrief/pepys-import#817]
```
When I load up the Privacy table, with the intention of deleting the Private and Public entries, the whole 
window freezes up and I will have to restart Pepys Admin.
```

I'll try to reproduce it.

Note: I've just tried to reproduce this on the 16Mb sqlite database.   I was able to load the privacies table, but when I try to delete it has frozen.  _Maybe_ it hasn't frozen, but is taking a long time to sort out the dependent fields.

Aah, it has returned after a couple of minutes.
![image](https://user-images.githubusercontent.com/1108513/110828817-57ce3c80-828f-11eb-97f0-b6a678027cb1.png)

**Comment by robintw** [comment]
I can't seem to reproduce this either. I suspect I may not have tested fully with the Privacies table, but it seems to work ok for me - it loads the table, I can then filter normally and edit successfully.

There is code in place that should catch any exceptions that occur during the creation of the column data for a table - which is what happens when you select a table from the dropdown on the top left. I'm intrigued that it actually hangs, rather than giving an exception of some sort.

Can you ask them whether the progress bar saying `Loading table data` appears, and roughly how far along the progress bar it gets before it crashes?

**Comment by IanMayo** [comment]
Issue description updated, @robintw

**Comment by robintw** [comment]
Ahh yes, freezing when they try and delete a privacy that is referenced in a lot of places makes more sense. I'll add a progress bar to the dependent objects calculation tomorrow - it'll be a bit tricky to work out how to increment the progress bar, as the time consuming thing is finding the objects, and so we don't know how many we have at that stage (ie. what the denominator would be for the proportion completed). I _think_ the best thing to do will be to have the progress bar go round and round (going 0 to 100, then back to 0 to 100 etc), as then at least it will be continually moving and showing that something is happening.

**Comment by IanMayo** [comment]
I guess the denominator is the number of rows in that table.  The # of
row in the tables are probably:
* platforms: 50
* datafiles: 600
* states: 1.5m
* comments: 5k

So the denominator for the above would be ~ 1.505m, as we process
platforms numerator goes to 50, then 650, then 1.5m, then 1.505m (I
guess...)

On Thu, 11 Mar 2021 at 17:36, Robin Wilson ***@***.***> wrote:
>
> Ahh yes, freezing when they try and delete a privacy that is referenced in a lot of places makes more sense. I'll add a progress bar to the dependent objects calculation tomorrow - it'll be a bit tricky to work out how to increment the progress bar, as the time consuming thing is finding the objects, and so we don't know how many we have at that stage (ie. what the denominator would be for the proportion completed). I think the best thing to do will be to have the progress bar go round and round (going 0 to 100, then back to 0 to 100 etc), as then at least it will be continually moving and showing that something is happening.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.



-- 
Ian Mayo
Deep Blue C Technology Ltd
- New technologies, implemented well

**Comment by IanMayo** [comment]
Note: _maybe_ this is an indicator that we need additional indexes on the database.

**Comment by robintw** [comment]
Fixed in #819.
